### PR TITLE
Fix plugin fails to configure android projects with flavors.

### LINKED ISSUE #131: Compatibility issue with Android tools 3.3.0-alpha11
I just tried the `6.0.0` version of ktlint Gradle plugin with the latest version of Android Gradle plugin. 

The latest version of ktlint removed the `afterEvaluate`
The latest version of AGP introduced lazy task creation. 

When they come together, I got the following error: 

> Cannot add task ‘assembleDebug’ as a task with that name already exists.

<details>
Caused by: org.gradle.api.internal.tasks.DefaultTaskContainer$DuplicateTaskException: Cannot add task 'assembleDebug' as a task with that name already exists.
    at org.gradle.api.internal.tasks.DefaultTaskContainer.duplicateTask(DefaultTaskContainer.java:250)
    at org.gradle.api.internal.tasks.DefaultTaskContainer.registerTask(DefaultTaskContainer.java:365)
    at org.gradle.api.internal.tasks.DefaultTaskContainer.register(DefaultTaskContainer.java:345)
    at com.android.build.gradle.internal.tasks.factory.TaskFactoryImpl.register(TaskFactoryImpl.kt:84)
    at com.android.build.gradle.internal.TaskManager.createAnchorAssembleTasks(TaskManager.java:3038)
    at com.android.build.gradle.BasePlugin.createAndroidTasks(BasePlugin.java:814)
    at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:81)
    at com.android.build.gradle.BasePlugin.lambda$createTasks$4(BasePlugin.java:667)
    at com.android.build.gradle.internal.crash.CrashReporting$afterEvaluate$1.execute(crash_reporting.kt:37)
    at com.android.build.gradle.internal.crash.CrashReporting$afterEvaluate$1.execute(crash_reporting.kt)
</details>

AGP tries to `register` the task but the `assembleDebug` task is already added by `ktlint` Gradle plugin. Not sure if `ktlint` really creates a task named `assembleDebug` but it would worth double-checking. 

Reverting back to `5.0.0` fixes the problem. :+1:

### PR DESCRIPTION
**Pull request by Tapchicoma**
Rethink approach how plugin gets android project sources and now it
uses android plugins base extension information, rather then VariantManager.

This also removes check of the same code more then once.

Closes #131

### CODE REVIEW

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt [hunk 1]
```diff
@@ -91,56 +85,50 @@ open class KtlintPlugin : Plugin<Project> {
...
-            target.plugins.withId("com.android.feature") { plugin ->
-                (plugin as FeaturePlugin).variantManager.run(createTasks)
+                addKtlintFormatTaskToProjectMetaFormatTask(target, ktlintSourceSetFormatTask)
             }
-            target.plugins.withId("com.android.test") { plugin ->
-                (plugin as TestPlugin).variantManager.run(createTasks)
+
+            // Variant manager returns all sources for variant,
+            // so most probably main source set maybe checked several times.
+            // This approach creates one check tasks per one source set.
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt]: Multiline comment instead?

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt [hunk 2]
```diff
@@ -91,56 +85,50 @@ open class KtlintPlugin : Plugin<Project> {
...
+            val pluginConfigure: (Plugin<Any>) -> Unit = { _ ->
+                target.extensions.configure(BaseExtension::class.java) { ext ->
+                    ext.sourceSets { sourceSet ->
+                        sourceSet.all {
+                            createTasks(it.name, target.files(it.java.srcDirs))
+                        }
+                    }
+                }
             }
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt]: Why not make this a function? This is fine if it's just straight up simpler.

File: samples/android-app/build.gradle.kts [hunk 3]
```diff
@@ -26,6 +26,12 @@ configure<AppExtension> {
         }
     }
 
+    flavorDimensions("beer")
+    productFlavors {
+        register("weissbier")
+        register("kellerbier")
+    }
```
Comment (JLLeitschuh) [inline samples/android-app/build.gradle.kts]: Can you explain this?

Comment (Tapchicoma) [reply samples/android-app/build.gradle.kts]: It is android build variants. 

For example, in this case, app can include special sources/resources that only belongs to `"weissbier"` variant on top of main sources/resources.

Such approach allows to have multiple variants of the same app, like special variant for play store and variant of the app for F-Droid store.

Comment (JLLeitschuh) [reply samples/android-app/build.gradle.kts]: Cool, good to know!

**Comment by Tapchicoma** [comment]
@JLLeitschuh addressed comments

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt [hunk 4]
```diff
@@ -111,10 +111,12 @@ open class KtlintPlugin : Plugin<Project> {
...
 
-            // Variant manager returns all sources for variant,
-            // so most probably main source set maybe checked several times.
-            // This approach creates one check tasks per one source set.
-            val pluginConfigure: (Plugin<Any>) -> Unit = { _ ->
+            /*
+            Variant manager returns all sources for variant,
+            so most probably main source set maybe checked several times.
+            This approach creates one check tasks per one source set.
+            */
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt]: This is more what I meant:
```kotlin
            /*
             * Variant manager returns all sources for variant,
             * so most probably main source set maybe checked several times.
             * This approach creates one check tasks per one source set.
             */
```

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt [hunk 5]
```diff
@@ -124,11 +126,11 @@ open class KtlintPlugin : Plugin<Project> {
...
-            target.plugins.withId("com.android.application", pluginConfigure)
-            target.plugins.withId("com.android.library", pluginConfigure)
-            target.plugins.withId("com.android.instantapp", pluginConfigure)
-            target.plugins.withId("com.android.feature", pluginConfigure)
-            target.plugins.withId("com.android.test", pluginConfigure)
+            target.plugins.withId("com.android.application", getPluginConfigureAction())
+            target.plugins.withId("com.android.library", getPluginConfigureAction())
+            target.plugins.withId("com.android.instantapp", getPluginConfigureAction())
+            target.plugins.withId("com.android.feature", getPluginConfigureAction())
+            target.plugins.withId("com.android.test", getPluginConfigureAction())
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt]: Uhh. Sure. this works 😆 Thanks!

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt (hunk 4 above)
Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt]: Changed to this :slightly_smiling_face:

**Pull request merged**